spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/config/ConfigFileApplicationListenerTests.java

```diff
@@ -1150,6 +1151,46 @@ void locationsWithWildcardFilesShouldIgnoreHiddenDirectories() {
 		assertThat(this.environment.getProperty("fourth.property")).isNull();
 	}
 
+	@Test
+	void additionalProfilesCanBeIncludedFromProgrammaticallySetting() {
+		// gh-25704
+		SpringApplication application = new SpringApplication(Config.class);
+		application.setWebApplicationType(WebApplicationType.NONE);
+		application.setAdditionalProfiles("dev");
+		this.context = application.run();
+		// Active profile should win over default
+		assertThat(this.context.getEnvironment().getProperty("my.property")).isEqualTo("fromdevpropertiesfile");
+	}
+
+	@Test
+	void twoAdditionalProfilesCanBeIncludedFromProgrammaticallySetting() {
+		// gh-25704
+		SpringApplication application = new SpringApplication(Config.class);
+		application.setWebApplicationType(WebApplicationType.NONE);
+		application.setAdditionalProfiles("other", "dev");
+		this.context = application.run();
+		assertThat(this.context.getEnvironment().getProperty("my.property")).isEqualTo("fromdevpropertiesfile");
+	}
+
+	@Test
+	void includeProfilesOrder() {
+		SpringApplication application = new SpringApplication(Config.class);
+		application.setWebApplicationType(WebApplicationType.NONE);
+		this.context = application.run("--spring.profiles.active=bar,spam", "--spring.profiles.include=foo");
+		// Before Boot 2.4 included profiles should always be first
+		assertThat(this.context.getEnvironment().getActiveProfiles()).containsExactly("foo", "bar", "spam");
```

Comment:
This test isn't related to the bug. However, I think that we are lacking test cases of the order included profiles and active profiles. So, Could I add this test to another PR?

Because of lacking test cases of the order additional profiles and active profiles, it results in https://github.com/spring-projects/spring-boot/issues/26189

Reply:
This doesn't seem to be testing active and additional profiles though. It's testing active and included profiles.

Reply:
@mbhave 
https://github.com/spring-projects/spring-boot/pull/25817#discussion_r626168473
As your above comment, there is already a test case that tests active and included profiles when using legacy processing, so I will delete this test case.